feat: added a deploy script

Beardrop2 opened the pull request:
The deploy script should install, test, and run the bot.

Beardrop2 committed:
feat: added a deploy script

The deploy script should install, test, and run the bot.

interrrp commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,35 @@
+    runs-on: self-hosted
+    needs: test
+    steps:
+      - run: python ./bot/__main__.py
```
Should use `pdm start` or `python -m bot` here

Beardrop2 replied:
Yeah will be python -m bot, but until we setup the bot dependency this will proc the hello world output.

Beardrop2 committed:
fix: Consolidate deploy

"uses" key uses docker images.
KISS, run in one job.

Beardrop2 committed:
feat: Pytest

Added pytest

Beardrop2 committed:
fix: distribution on

Need to toggle the dist so that the CI can run pdm build

Beardrop2 committed:
feat: Start and stop scripts

Added start and stop scripts that won't time out. Will probably replace with docker compose

Beardrop2 committed:
fix: update version and authors

Set the version to 0.1.0 as the first minor change to an unreleased version.
Added myself as an author.

interrrp commented on pyproject.toml:
```diff
@@ -50,5 +59,18 @@ ignore = [
+
 [tool.pdm]
-distribution = false
+distribution = true
```
We can just remove the `tool.pdm` section since `distribution = true` is the default

interrrp commented on pyproject.toml:
```diff
@@ -1,14 +1,23 @@
 
 [tool.pdm.scripts]
-start = { cmd = "python -m bot" }
+test = {cmd = "python -m pytest"}
```
Inconsistent formatting (no spaces around table content)

interrrp commented on pyproject.toml:
```diff
@@ -1,14 +1,23 @@
 [project]
 name = "zoomy-zodiacs"
-version = "0"
+authors = [
```
Should add the other team members

Beardrop2 committed:
fix: Added team names

I don't have emails, and where I don't have a github name, I used the discord name.

Beardrop2 committed:
fix: formating of test cmd
